src/core/src/op/scaled_dot_product_attention.cpp

```diff
@@ -44,14 +44,14 @@ void op::v13::ScaledDotProductAttention::validate_and_infer_types() {
     const auto& causal = get_causal();
     if (input_size >= 4 && !causal) {
         const auto& attention_type = get_input_element_type(3);
-        NODE_VALIDATION_CHECK(
-            this,
-            attention_type.is_real() || attention_type == element::boolean || attention_type.is_dynamic(),
-            "The element type of attention_mask must be either floating-point or boolean.");
+        NODE_VALIDATION_CHECK(this,
```

Comment:
It is not according operator's specification as the input should be any supported floating-point type.

@mitruska 
Should be specification updated? Is possible to change the input this input type?